### ISSUE TITLE
Add version info when register seata-server address to the registry.

### ISSUE THREAD

**Issue by ujjboy** [seata/seata#894]
`## Why you need it?

Now, when register `seata-server` address to the registry, it only contains `IP`,`port` and `cluster`, but no `version` info. 

For better scalability and compatibility in the future, it is recommended to bring the version number information.

## How it could be?
- Add version info when do register.
- if version from the registry is null, means old version.

## Other related information
#893

**Comment by xingfudeshi** [comment]
good.

**Comment by slievrly** [comment]
RegisterRMRequest，RegisterTMRequest contains version for a channel，the server can verify the version of the client.

**Comment by ujjboy** [comment]
Yes, but the Client doesn't know the version of Server before connect it.

After add verison, this feature can be used to be compatible with the following scenarios:
- when a new client (supporting old and new protocol) connect to old server (only supports old protocol)
